docs/reference/scripting/painless.asciidoc

```diff
@@ -1,32 +1,35 @@
 [[modules-scripting-painless]]
 == Painless scripting language
 
-_Painless_ is a simple, secure scripting language designed specifically for use
-with Elasticsearch. It is the default scripting language for Elasticsearch and
-can safely be used for inline and stored scripts. To get started with
-Painless, see the {painless}/painless-guide.html[Painless Guide]. For a
-detailed description of the Painless syntax and language features, see the
-{painless}/painless-lang-spec.html[Painless Language Specification].
+_Painless_ is a performant, secure scripting language designed specifically for
+{es}. You can use Painless anywhere scripts are supported in {es} to safely
+write inline and stored scripts.
 
 [[painless-features]]
-You can use Painless anywhere scripts can be used in Elasticsearch. Painless
-provides:
-
-* Fast performance: Painless scripts https://benchmarks.elastic.co/index.html#search_qps_scripts[
-run several times faster] than the alternatives.
-
-* Safety: Fine-grained allowlists with method call/field granularity. See the
-{painless}/painless-api-reference.html[Painless API Reference] for a
-complete list of available classes and methods.
-
-* Optional typing: Variables and parameters can use explicit types or the
-dynamic `def` type.
-
-* Syntax: Extends a subset of Java's syntax to provide additional scripting
-language features.
-
-* Optimizations: Designed specifically for Elasticsearch scripting.
-
-Ready to start scripting with Painless? See the
-{painless}/painless-guide.html[Painless Guide] for the
-{painless}/index.html[Painless Scripting Language].
\ No newline at end of file
+Painless provides numerous capabilities that center around the following
+core principles:
+
+* **Safety**: Ensuring the security of your cluster is of utmost importance. To
+that end, Painless uses a fine-grained allowlist with a granularity down to the
+members of a class. Anything that is not part of the allowlist simply does not
```

Comment:
Trying to use something outside of the whitelist is a compilation error.  "does not execute" implies it's been compiled.

Reply:
++

I'll change to read: "Anything that is not part of the allowlist results in a compilation error."